integration/src/main/java/scripts/testscripts/DatasetOperations.java

```diff
@@ -173,43 +173,46 @@ private void assertDatasetValues(
     keys.forEach(key -> assertThat(jsonOne.get(key), is(expectedMetadata.get(key))));
   }
 
-  private void crudUserJourney(CatalogClient client, StorageSystem storageSystem, String sourceId)
-      throws Exception {
+  private void crudUserJourney(StorageSystem storageSystem, String sourceId) throws Exception {
     // Given a snapshot, create a catalog entry.
     var request =
         new CreateDatasetRequest()
             .catalogEntry(createMetadata("crud").toString())
             .storageSourceId(sourceId)
             .storageSystem(storageSystem);
-    datasetId = datasetsApi.createDataset(request).getId();
-    assertThat(client.getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+    var createResponse = datasetsApi.createDatasetWithHttpInfo(request);
```

Comment:
How come this is `.createDatasetWithHttpInfo` rather than `createDataset`? Did the operation id change? Or is this a new method that returns the full http object instead of just the response body?

Reply:
Yes, although the library is the same (`jersey2`) the client codegen is still different. One difference is that the HTTP code isn't present in `client` anymore. So instead we need to use the `.xxxWithHttpInfo()` API variants.